libkineto/src/output_json.cpp

```diff
@@ -244,6 +244,50 @@ void ChromeTraceLogger::handleGenericActivity(
       op.traceSpan()->name, op.traceSpan()->iteration, separator,
       op_metadata);
   // clang-format on
+
+  if (op.linkedActivity() != nullptr) {
+    handleGenericLink(op);
+  }
+}
+
+void ChromeTraceLogger::handleGenericLink(const GenericTraceActivity& op) {
+  uint64_t link_type = GET_LINK_TYPE(op.linkId);
+  if (link_type == LINK_FORWARD_BACKWARD_HEAD) {
```

Comment:
Assuming linkedActivity is set only for destination and above bitfield suggestion:

if (op.flow.linkType == kLinkFwdBwd) {
  handleGenericLinkStart(op.flow.linkedActivity(), "forward_backward", "fwd_bwd");
  handleGenericLinkEnd(op, "forward_backward", "fwd_bwd");
}